src/main/java/org/apache/sysml/hops/UnaryOp.java

```diff
@@ -563,6 +569,10 @@ protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz
 			// getMemEstimate works for both cases of known dims and worst-case stats
 			ret = getInput().get(0).getMemEstimate() * 3; 
 		}
+
+		if (DMLScript.USE_ACCELERATOR) {
+			OptimizerUtils.estimateSize(dim1, dim2); // Intermediate memory required to convert sparse to dense
+		}
 		
 		return ret;
 	}
```

Comment:
Well, I just find it suspicious that the ConvolutionOp has intermediate memory requirements of zero although you just assigned memory intermediates to matrix multiplications. This is probably something that needs fixing in general (also for CP).